WordPress/WordPressTest/ReaderCardServiceTests.swift

```diff
@@ -0,0 +1,98 @@
+import UIKit
+import XCTest
+import Nimble
+
+@testable import WordPress
+
+class ReaderCardServiceTests: XCTestCase {
+
+    private var coreDataStack: CoreDataStack!
+    private var remoteService: ReaderPostServiceRemote!
+    private var apiMock: WordPressComMockRestApi!
+
+    override func setUp() {
+        super.setUp()
+        coreDataStack = TestContextManager()
+        apiMock = WordPressComMockRestApi()
+        remoteService = ReaderPostServiceRemote(wordPressComRestApi: apiMock)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        ContextManager.overrideSharedInstance(nil)
+    }
+
+
+    /// Save 9 cards in the database
+    /// The API returns 10, but one of them is unknown and shouldn't be saved
+    ///
+    func testSaveCards() {
+        let expectation = self.expectation(description: "Image should fail to return a video asset.")
+
+        let service = ReaderCardService(service: remoteService, coreDataStack: coreDataStack)
+        apiMock.succeed = true
+
+        service.fetch(success: { _, _ in
+            let cards = try? self.coreDataStack.mainContext.fetch(NSFetchRequest(entityName: ReaderCard.classNameWithoutNamespaces()))
+            expect(cards?.count).to(equal(9))
+            expectation.fulfill()
+        }, failure: { _ in })
+
+        waitForExpectations(timeout: 5, handler: nil)
+    }
+
+    /// From the 9 cards saved, 8 should have posts
+    ///
+    func testSaveCardsWithPosts() {
+        let expectation = self.expectation(description: "Image should fail to return a video asset.")
```

Comment:
Same as above

Reply:
Done in 408b6d055836bb5af6cbe5e6b88015334dd3f820 (I also skipped the tests since the CD model changes weren't commited)